### PR TITLE
Add Ethernet support on mr_canhubk3 board

### PR DESCRIPTION
**Pull request by manuargue**
Introduce Ethernet shim driver for NXP S32 GMAC/EMAC:
- it's a copy-based implementation with DMA data buffers and buffer descriptors in non-cached memory (buf len and ring size configurable)
- no PHY driver integration for the moment, it works as a fixed link with speed/duplex configured through devicetree (ie. relies on the external PHY to set-up and perform auto-negotiation based on pin strapping configuration).
- supports multicast hash filtering, promiscuous mode, MAC loopback
~- supports MDIO STA for both Clause 22 and Clause 45. Clause 22 access is supported through the MDIO API and
Clause 45 through device-specific APIs, due to a limitation of current MDIO API which doesn't include a parameter for the MMD in the RW operations.~

This board has a single instance of EMAC connected to a NXP TJA1103 Ethernet PHY, for which there's no currently driver.

### CODE REVIEW

**Comment by manuargue** [comment]
Hi @tbursztyka , @rlubos - and perhaps @danieldegrasse. Pinging you to please have a look to these changes, your feedback is more than welcome. Thanks

File: drivers/ethernet/eth_nxp_s32_gmac.c [hunk 1]
```diff
@@ -0,0 +1,788 @@
...
+
+#if defined(ETH_NXP_S32_MULTICAST_FILTER)
+static void eth_nxp_s32_mcast_cb(struct net_if *iface, const struct net_addr *addr, bool is_joined)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct eth_nxp_s32_data *ctx = dev->data;
+	struct net_eth_addr mac_addr;
+
+	switch (addr->family) {
+#if defined(CONFIG_NET_NATIVE_IPV4)
```
[the remark this answers is not in the document]
Comment (manuargue) [reply drivers/ethernet/eth_nxp_s32_gmac.c]: right, this is an error, it was intended `CONFIG_NET_IPV*`

**Comment by dleach02** [comment]
@decsny please help review this PR

**Comment by decsny** [comment]
@manuargue I am a bit confused as to where the MDIO driver added in this PR is going to be used, since there is no phy driver being added for the phy on the board being enabled here, and the main ethernet driver relies on pin strapping for the connected phy.

**Comment by manuargue** [comment]
> @manuargue I am a bit confused as to where the MDIO driver added in this PR is going to be used, since there is no phy driver being added for the phy on the board being enabled here, and the main ethernet driver being added for the SOC relies on pin strapping of the connected phy.

@decsny the initial idea was to advance work and enable as much as possible for the GMAC peripheral, but you got a point. We'll wait to introduce the MDIO support once there's a PHY driver, and perform the necessary updates in the Eth driver to integrate both. So I have dropped MDIO from this PR.

**Review by decsny — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@manuargue, kindly rebase and update `west.yml`.

**Review by decsny — DISMISSED** [review]

**Comment by manuargue** [comment]
@decsny I have rebased to fix conflicts

**Review by decsny — APPROVED** [review]

File: drivers/ethernet/eth_nxp_s32_gmac.c [hunk 2]
```diff
@@ -0,0 +1,777 @@
...
+	if (clk_status != CLOCK_IP_SUCCESS) {
+		LOG_ERR("Failed to configure clocks (%d)", clk_status);
+		return -EIO;
+	}
+
+	/*
+	 * PHY mode selection must be done before the controller is reset,
+	 * because the interface type is latched at controller's reset
+	 */
+	err = select_phy_interface(cfg->ctrl_cfg.Gmac_pCtrlConfig->MiiMode);
```
Comment (mmahadevan108) [inline drivers/ethernet/eth_nxp_s32_gmac.c:137]: Should this be wrapped with the check for `CONFIG_SOC_PART_NUMBER_S32K3`

Comment (manuargue) [reply drivers/ethernet/eth_nxp_s32_gmac.c:137]: The phy interface selection must be done always independently of the SoC. Atm the only implementation provided is for K344, but later can be extended when more devices are supported.

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**